tiledb/cloud/dag/dag.py

```diff
@@ -368,17 +425,33 @@ def execute_update_callbacks(self):
         for func in self.update_callbacks:
             func(self)
 
-    def execute_done_callbacks(self):
-        """
-        Run user specified callbacks for DAG completion
-        :return:
-        """
+    def _report_completion(self) -> None:
+        """Reports the completion of the task graph to the server and callbacks."""
         if not self.called_done_callbacks:
+            self._report_server_completion()
             for func in self.done_callbacks:
                 func(self)
 
         self.called_done_callbacks = True
 
+    def _report_server_completion(self) -> None:
+        if not self.server_graph_uuid:
+            return
+        try:
+            api_st = _API_STATUSES[self.status]
+        except KeyError as ke:
+            raise ValueError(
+                f"Task graph ended in invalid state {self.status!r}"
+            ) from ke
+        try:
+            client.client.task_graph_logs_api.update_task_graph_log(
+                id=str(self.server_graph_uuid),
```

Comment:
After running a test against staging, I had to make one more change here, to add the `str` conversion. The generated client doesn’t like trying to serialize `uuid.UUID`s.